parser.ts

```diff
@@ -79,6 +80,35 @@ function wrap_locs<T extends Function>(traverser: T, storeSrc: boolean = false):
 export const traverseLiteral = wrap_locs(traverseLiteralHelper);
 export function traverseLiteralHelper(c: TreeCursor, s: string, env: ParserEnv): Literal<Annotation> {
   switch (c.type.name) {
+    case "String":
+      const str_ = s.substring(c.from+1,c.to-1);
+      console.log("str_");
+      let charArr = []
+      for (let i = 0; i < str_.length; i++) {
+        if (str_[i] != "\\"){
+          charArr.push(str_[i])
+        }
+        else {
+          i += 1
+          if (i == str_.length-1) {
+            throw new Error("Token Unrecognized" + "\\")
+          }
+          else if (str_[i] != "t" && str_[i] != "n" && str_[i] != "\\" && str_[i] != "\"") {
+            throw new Error("Token Unrecognized" + "\\" + str_[i])
+          }
+          else {
+            charArr.push(str_[i-1] + str_[i])
+            console.log("push")
+            console.log(charArr[charArr.length-1])
+          }
+        }
+      }
+      return{
+        tag: "str",
+        value: charArr,
+        length: charArr.length
```

Comment:
Does `charArr` ever change length once created? Why store the length separately?